src/Operation/Watch.php

```diff
@@ -104,79 +139,145 @@ public function __construct(Manager $manager, $databaseName, $collectionName, ar
             throw InvalidArgumentException::invalidType('"fullDocument" option', $options['fullDocument'], 'string');
         }
 
-        if (isset($options['resumeAfter'])) {
-            if ( ! is_array($options['resumeAfter']) && ! is_object($options['resumeAfter'])) {
-                throw InvalidArgumentException::invalidType('"resumeAfter" option', $options['resumeAfter'], 'array or object');
-            }
+        if (isset($options['resumeAfter']) && ! is_array($options['resumeAfter']) && ! is_object($options['resumeAfter'])) {
+            throw InvalidArgumentException::invalidType('"resumeAfter" option', $options['resumeAfter'], 'array or object');
         }
 
+        if (isset($options['startAtOperationTime']) && ! $options['startAtOperationTime'] instanceof TimestampInterface) {
+            throw InvalidArgumentException::invalidType('"startAtOperationTime" option', $options['startAtOperationTime'], 'MongoDB\BSON\TimestampInterface');
+        }
+
+        /* In the absence of an explicit session, create one to ensure that the
+         * initial aggregation and any resume attempts can use the same session
+         * ("implicit from the user's perspective per PHPLIB-342). We can ignore
+         * the exception, as libmongoc likely cannot create an implicit session
+         * either and there is no risk of a mismatch. */
         if ( ! isset($options['session'])) {
             try {
                 $options['session'] = $manager->startSession();
-            } catch (DriverRuntimeException $e) {}
+            } catch (RuntimeException $e) {}
+        }
+
+        $this->aggregateOptions = array_intersect_key($options, ['batchSize' => 1, 'collation' => 1, 'maxAwaitTimeMS' => 1, 'readConcern' => 1, 'readPreference' => 1, 'session' => 1, 'typeMap' => 1]);
+        $this->changeStreamOptions = array_intersect_key($options, ['fullDocument' => 1, 'resumeAfter' => 1, 'startAtOperationTime' => 1]);
+
+        // Null database name implies a cluster-wide change stream
+        if ($databaseName === null) {
+            $databaseName = 'admin';
+            $this->changeStreamOptions['allChangesForCluster'] = true;
         }
 
         $this->databaseName = (string) $databaseName;
-        $this->collectionName = (string) $collectionName;
+        $this->collectionName = isset($collectionName) ? (string) $collectionName : null;
```

Comment:
`Aggregate` takes care of handling the `null` Collection name (per [PHPLIB-358](https://jira.mongodb.org/browse/PHPLIB-358)).